packages/providers/test/provider.test.ts

```diff
@@ -24,6 +24,11 @@ afterEach(() => {
   jest.restoreAllMocks();
 });
 
+const resetCache = () => {
+  Provider.chainInfoCache = {};
+  Provider.nodeInfoCache = {};
```

Comment:
I've opened up #1295. If you could kill two birds with one stone and close that issue here as well - great. If not, no problem.

Reply:
@luizstacio @LuizAsFight In my first review with @ Dhaiwat10, I remember there was a method to refresh/invalidate the cache, but it was later removed in favor of two different methods.

I have never implemented cache in this fashion as well:
 - `getXYZ` — get XYZ from the cache
 - `fetchXYZ` — fetch XYZ remotely and update the cache

We usually have a single method that will transparently decide if it gets stuff from the cache or remotely. Then, we may also provide a way for resetting, refreshing, or invalidating the cache _(whatever floats your boat)_.

Reply:
When dealing with cache is a good way to have cache keys, and invalidation actions instead of multiple methods.

It would be better if we could invalidate specific parts of the cache. Or having options that can be provided to all methods, fetchXYX(...params, options: { invalidate: bool }).

These are just some ideas but not a specific way, on how to do it.

Reply:
@arboleya you may not like the fashion and do it in other way, as long as it supports what we need. the code delivered before didn't support.

having a single method that will decide, will need to be async, which is not acceptable in our use case. that's why I go a bit different way that will make more sense for our scenario of memoization.

`get/fetch` is the most intuitive way I imagined

Reply:
There seems to be more depth to this issue than I initially imagined when writing the comment. I'd move the conversation to #1295. @LuizAsFight could you please share your use cases (with code references) on that issue so that we get the full picture?